app/src/main/java/com/zulip/android/networking/AsyncGetEvents.java

```diff
@@ -507,9 +509,17 @@ private void processUpdateMessages(List<EventsBranch> updateEvents) {
             @Override
             public void run() {
                 RecyclerMessageAdapter adapter = mActivity.getCurrentMessageList().getAdapter();
+                int i = 0;
                 for (int id : messageIds) {
                     // notify adapter data item changed
-                    adapter.notifyItemChanged(adapter.getItemIndex(id));
+                    int index = adapter.getItemIndex(id);
+                    adapter.notifyItemChanged(index);
+                    RecyclerView.ViewHolder viewHolderForAdapterPosition = mActivity.getCurrentMessageList().getRecyclerView().findViewHolderForAdapterPosition(index);
+                    if (viewHolderForAdapterPosition instanceof MessageHolder) {
+                        UpdateMessageWrapper updateEvent = (UpdateMessageWrapper) updateEvents.get(i++);
```

Comment:
I have added a check https://github.com/zulip/zulip-android/pull/408/files#diff-12a339ef893658709ac106c1e3a45a5eR516 so if there doesn't exists a message of that ID in the list it will skip that message!